src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2146,10 +2225,22 @@ private MemberDeclarationSyntax ParseMemberDeclarationOrStatementCore(SyntaxKind
 
                         if (IsFieldDeclaration(isEvent: false))
                         {
-                            if (acceptStatement)
+                            if (isGlobal && !haveAttributes && !(haveModifiers && IsScript))
```

Comment:
fwiw, these conditions are pretty hard to both grok and to determine correctness around.  i don't really get a sense for why this is the appropriate rules for this sort of situation.

Reply:
> fwiw, these conditions are pretty hard to both grok and to determine correctness around.  i don't really get a sense for why this is the appropriate rules for this sort of situation.

The conditions are mostly preserving original logic. ```acceptStatement``` was false if there were attributes or modifiers. We allow modifiers for non-script case though

---
In reply to: [379669424](https://github.com/dotnet/roslyn/pull/41706#discussion_r379669424) [](ancestors = 379669424)